packages/flutter/test/widgets/widget_inspector_structure_error_test.dart

```diff
@@ -13,82 +13,35 @@ void main() {
   StructureErrorTestWidgetInspectorService.runTests();
 }
 
-typedef InspectorServiceExtensionCallback = FutureOr<Map<String, Object?>> Function(Map<String, String> parameters);
-
 class StructureErrorTestWidgetInspectorService extends Object with WidgetInspectorService {
-  final Map<String, InspectorServiceExtensionCallback> extensions = <String, InspectorServiceExtensionCallback>{};
-
-  final Map<String, List<Map<Object, Object?>>> eventsDispatched = <String, List<Map<Object, Object?>>>{};
-
-  @override
-  void registerServiceExtension({
-    required String name,
-    required FutureOr<Map<String, Object?>> Function(Map<String, String> parameters) callback,
-  }) {
-    assert(!extensions.containsKey(name));
-    extensions[name] = callback;
-  }
-
-  @override
-  void postEvent(String eventKind, Map<Object, Object?> eventData) {
-    getEventsDispatched(eventKind).add(eventData);
-  }
-
-  List<Map<Object, Object?>> getEventsDispatched(String eventKind) {
-    return eventsDispatched.putIfAbsent(eventKind, () => <Map<Object, Object?>>[]);
-  }
-
-  Iterable<Map<Object, Object?>> getServiceExtensionStateChangedEvents(String extensionName) {
-    return getEventsDispatched('Flutter.ServiceExtensionStateChanged')
-      .where((Map<Object, Object?> event) => event['extension'] == extensionName);
-  }
-
-  Future<String> testBoolExtension(String name, Map<String, String> arguments) async {
-    expect(extensions, contains(name));
-    // Encode and decode to JSON to match behavior using a real service
-    // extension where only JSON is allowed.
-    // ignore: avoid_dynamic_calls
-    return json.decode(json.encode(await extensions[name]!(arguments)))['enabled'] as String;
-  }
-
-
   static void runTests() {
     final StructureErrorTestWidgetInspectorService service = StructureErrorTestWidgetInspectorService();
     WidgetInspectorService.instance = service;
 
-    test('ext.flutter.inspector.structuredErrors reports error to _structuredExceptionHandler on error', () async {
+    test('ext.flutter.inspector.structuredErrors - FlutterError.onError is called', () async {
```

Comment:
Removed stuff that was moved into `widget_inspector_test_utils.dart` and changed the test to test the proper behavior